Label 추가 컴포넌트 구현

ahrimy opened the pull request:
- Label List Page 에 새로운 Label을 생성하는 컴포넌트 추가
- New Label 버튼에 의해 Input 컴포넌트 toggle
- 초기값으로 color 값 랜덤 생성
- color refresh 버튼 클릭 시 랜덤으로 색 변경
- name, color 변경 시 preview 에 적용
- name 이 비어있거나, color가 유효하지 않은경우 submit 버튼 disabled
  - color가 유효하지 않으면 글씨 빨간색
- submit 버튼에 클릭 이벤트 추가
  - 서버에 데이터 전송 후 success 되면 새로고침
<img width="1440" alt="Screen Shot 2020-11-12 at 9 48 14 AM" src="https://user-images.githubusercontent.com/26592306/98880947-3a00ee80-24cc-11eb-9c6e-dde23d51e6f7.png">

Commit:
[client] feat: util 추가 및 svg 추가

- isValidColor: color 유효성 검사
- getRandomColor: 랜덤으로 색상 결정
- refreshIcon: refresh 버튼의 아이콘

Commit:
[client] feat: label input 컴포넌트 구현

- label input 컴포넌트 추가
- label input 컴포넌트를 구성하는 input 컴포넌트, cancel/submit 버튼 컴포넌트 추가
- input 값에 따라서 preview 변경 이벤트 구현
- refresh 버튼으로 새로운 랜덤 색 결정

Commit:
[client] feat: Label list 에 input 컴포넌트 추가

- Label List Page 에 Label Input 컴포넌트 추가
- New Label 버튼클릭시 Input 컴포넌트 toggle 되도록 하는 이벤트 추가

Commit:
[client] feat: submit 버튼 컴포넌트 생성

- input form 의 상태에 따라 submit 버튼의 disabled 결정하는 조건 추가
- submit button 클릭 시 post 요청 후 새로고침

Commit:
[client] feat: cancel 버튼 이벤트 추가

cancel 버튼 클릭 시 input view hidden

seokju2ng commented on web/client/src/components/common/GreenButton.js:
```diff
@@ -5,20 +5,21 @@ import COLOR from '../../utils/color';
-export default function GreenButton({ children }) {
-  return <Button>{children}</Button>;
+export default function GreenButton({ children, onClick }) {
+  return <Button onClick={e => onClick(e)}>{children}</Button>;
```
람다표현식이면 다음과 같이는 안되나요?
```js
<Button onClick={onClick}> ...</Button>
```

ahrimy replied:
음 그러게용....저런식으로 표현할 수 있을 것 같네요
이것도 반영해보겠습니당

seokju2ng commented on web/client/src/components/labelList/LabelInput/RefreshButton.js:
```diff
@@ -0,0 +1,29 @@
+
+export default function RefreshButton({ color, onClick }) {
+  return (
+    <Refresh {...{ color }} onClick={e => onClick(e)}>
```
여기 onClick도 위와 같이..

seokju2ng commented on web/client/src/utils/getRandomColor.js:
```diff
@@ -0,0 +1,4 @@
+const getRandomColor = () =>
+  (~~(Math.random() * 16777215)).toString(16).padEnd(6, '0');
```
padEnd가 아니라 padStart를 써야할 것 같아요
`FF`가 나오면 `#FF0000`가 되고 `FF0000`이 나와도 `#FF0000`이 되니까 원하는 범위의 랜덤이 제대로 동작하지 않을 것 같습니다.

ahrimy replied:
넵 `padStart`가 더 좋을거같네요!

seokju2ng reviewed (approved):
프론트 페이지 PR에는 스크린샷도 같이 올리면 좋을 것같아요 ㅠㅠ 
잘나왔는지 궁금한데 머지될 때까지 확인이 안되네용

skid901 reviewed (approved):
깔끔하게 작업해 주셔서 리뷰 드릴 부분이 없는 것 같아요!!
고생하셨습니다 ^^7

Commit:
[client] refactor: 리뷰 반영

- 컴포넌트에서 onClick함수 호출 방법 변경
- padEnd 대신 padStart 사용

Commit:
[client] fix: isValidColor와 isLightColor 판단기준 통일

- isValidColor에서는 6자리와 3자리를 유효하다고 판단하지만 isLightColor에서는 6자리만 유효하다고 판단하기 때문에 3자리일때 글씨 색이 변경되지 않는 버그 발생
- 6자리만 유효하도록 변경
